engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/serialization/serializer/ConfluentBaseKafkaAvroDeserializer.scala

```diff
@@ -0,0 +1,141 @@
+package pl.touk.nussknacker.engine.avro.schemaregistry.confluent.serialization.serializer
+
+import java.io.IOException
+import java.nio.ByteBuffer
+import java.util
+
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException
+import io.confluent.kafka.serializers.{AbstractKafkaAvroSerDe, KafkaAvroDeserializerConfig}
+import org.apache.avro.Schema
+import org.apache.avro.Schema.Type
+import org.apache.avro.generic.GenericDatumReader
+import org.apache.avro.io.{DatumReader, DecoderFactory}
+import org.apache.avro.reflect.ReflectDatumReader
+import org.apache.avro.specific.SpecificDatumReader
+import org.apache.kafka.common.errors.SerializationException
+import pl.touk.nussknacker.engine.avro.AvroUtils
+import pl.touk.nussknacker.engine.avro.schemaregistry.confluent.client.ConfluentSchemaRegistryClient
+
+/**
+  * @TODO: After update to newer version of confluent we should try extend by AbstractKafkaAvroDeserializer
+  *
+  * @param confluentSchemaRegistry
+  * @param isKey
+  */
+abstract class ConfluentBaseKafkaAvroDeserializer(confluentSchemaRegistry: ConfluentSchemaRegistryClient, var isKey: Boolean)
```

Comment:
If we make such large-copy paste maybe it's easier if we stick to java...

Reply:
I have already done this copy paste plus some specyfick modification to our two cases: STATIC / FRECH.. In future we will update confluent to newer version and then we remove all these copied methods.. Let's leave it now.

Reply:
IMO it is not good approach. Now you remember what you changed but after > 1year nobody will know and somone will introduce some regression bug assuming that there was no modification. Please add it as a java in one commit without change and in the second one add modifications that are need.

Reply:
We can say the same thing about formatter, and else stuff copied from Confluent..